components/code-editor/index.js

```diff
@@ -1,8 +1,84 @@
+/**
+ * External dependencies
+ */
+import { extend } from 'lodash';
+
 /**
  * WordPress dependencies
  */
 import { Component } from '@wordpress/element';
 
+/**
+ * Internal dependencies
+ */
+import withLazyDependencies from '../higher-order/with-lazy-dependencies';
+
+// TODO: How can we avoid repeating all of this?
+const defaultSettings = {
```

Comment:
How is it exposed for Plugins' editor?

Reply:
Normally these settings are assigned into `wp.codeEditor.defaultSettings` by `wp_enqueue_code_editor` here:

https://github.com/WordPress/wordpress-develop/blob/8f95800d52c1736d651ae6e259f90ad4a0db2c3f/src/wp-includes/general-template.php#L3576

But since the whole idea of this PR is to avoid calling `wp_enqueue_code_editor` (since it enqueues a lot of heavy assets), I've had to re-declare these editor settings here.

One possibility is that we could have Core expose the default settings separately e.g. with a new `wp_default_code_editor_settings()` method.

Any better ideas, @westonruter?

Reply:
@noisysocks We actually did have a separate function originally for getting the settings, but then combined them into `wp_enqueue_code_editor()`: https://github.com/WordPress/better-code-editing/commit/70b96cabc29a91f6f29cf55a56c8862a1f726bb5

Oh the irony.

Until a `wp_default_code_editor_settings()` is introduced, a workaround could be to do this:

```php
$gutenberg_editor_settings = array();
function gutenberg_capture_code_editor_settings_and_short_circuit_enqueues( $settings ) {
    global $gutenberg_editor_settings;
    $gutenberg_editor_settings = $settings;
    return false;
}
add_filter( 'wp_code_editor_settings', 'gutenberg_capture_code_editor_settings_and_short_circuit_enqueues', 1000 );
wp_enqueue_code_editor();
remove_filter( 'wp_code_editor_settings', 'gutenberg_capture_code_editor_settings_and_short_circuit_enqueues', 1000 );
```

Reply:
That's a fun hack!